services/webnn/ort/graph_builder_ort.cc

```diff
@@ -2220,79 +2261,183 @@ GraphBuilderOrt::AddLayerNormalizationOperation(
       }
       default:
         NOTREACHED() << "[WebNN] LayerNormalization only supports float32 "
                         "and float16 data type.";
     }
-
-    inputs.push_back(scale.c_str());
-  }
-
-  std::string bias;
-  if (layer_normalization.bias_operand_id) {
-    bias = GetOperandNameById(layer_normalization.bias_operand_id.value());
-    inputs.push_back(bias.c_str());
+    std::array<const char*, 2> add_inputs = {variance_output.data(),
+                                             epsilon_value.data()};
+    std::array<const char*, 1> add_outputs = {add_output.data()};
+    model_editor_.AddNode(kOpTypeAdd, inserted_node, add_inputs, add_outputs);
+
+    inserted_node = GenerateNextOperationName("inserted_sqrt");
+    const std::string sqrt_output = GenerateNextOperandName();
+    std::array<const char*, 1> sqrt_inputs = {add_output.data()};
+    std::array<const char*, 1> sqrt_outputs = {sqrt_output.data()};
+    model_editor_.AddNode(kOpTypeSqrt, inserted_node, sqrt_inputs,
+                          sqrt_outputs);
+
+    inserted_node = GenerateNextOperationName("inserted_div");
+    const std::string div_output = GenerateNextOperandName();
+    std::array<const char*, 2> div_inputs = {sub_output.data(),
+                                             sqrt_output.data()};
+    std::array<const char*, 1> div_outputs = {div_output.data()};
+    model_editor_.AddNode(kOpTypeDiv, inserted_node, div_inputs, div_outputs);
+
+    // Reshape to compatible_shape to match the input_shape for emulation
+    // calculation.
+    std::vector<uint32_t> compatible_shape(input_shape.size(), 1);
+    for (auto axis : axes) {
+      compatible_shape[axis] = input_shape[axis];
+    }
+    ASSIGN_OR_RETURN(scale, CreateOrTransposeScaleForLayerNomalization(
+                                layer_normalization.scale_operand_id,
+                                input_data_type, scale_shape, permutation));
+    ASSIGN_OR_RETURN(scale, PrependReshape(scale, compatible_shape));
+    if (layer_normalization.bias_operand_id) {
+      inserted_node = GenerateNextOperationName("inserted_mul");
+      const std::string mul_output = GenerateNextOperandName();
+      std::array<const char*, 2> mul_inputs = {scale.data(), div_output.data()};
+      std::array<const char*, 1> mul_outputs = {mul_output.data()};
+      model_editor_.AddNode(kOpTypeMul, inserted_node, mul_inputs, mul_outputs);
+
+      bias = GetOperandNameById(layer_normalization.bias_operand_id.value());
+      bias = PrependTranspose(bias, permutation);
+      ASSIGN_OR_RETURN(bias, PrependReshape(bias, compatible_shape));
+      std::array<const char*, 2> add_2_inputs = {mul_output.data(),
+                                                 bias.data()};
+      model_editor_.AddNode(kOpTypeAdd, node, add_2_inputs, outputs);
+    } else {
+      std::array<const char*, 2> mul_inputs = {scale.data(), div_output.data()};
+      model_editor_.AddNode(kOpTypeMul, node, mul_inputs, outputs);
```

Comment:
Should the inserted mul in `eslse` branch be consolidated with the one in the `if (layer_normalization.bias_operand_id) {` branch?